cmd/goose/main.go

```diff
@@ -83,7 +83,9 @@ func main() {
 	if *sequential {
 		goose.SetSequential(true)
 	}
-	goose.SetTableName(*table)
+
+	// Envvars should have lower priority than flags.
+	goose.SetTableName(firstNonEmpty(*table, envConfig.table))
```

Comment:
I don't think this will work, since `*table` will be set to the default from line 31:

```
flags.String("table", "goose_db_version", "migrations table name")
```

I think what might work here is:

```diff
- goose.SetTableName(firstNonEmpty(*table, envConfig.table))
+ goose.SetTableName(firstNonEmpty(*table, envConfig.table, "goose_db_version"))
```

And then remove the flag default:

```diff
- table        = flags.String("table", "goose_db_version", "migrations table name")
+ table        = flags.String("table", "", "migrations table name")
```

I believe that should maintain the order of precedence of flag > env > default.

Reply:
Thanks for looking into this @mfridman! I’ve made a change that should fix it. Please let me know if there are any other changes.